runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogDataSerializerUtils.java

```diff
@@ -0,0 +1,211 @@
+package org.corfudb.protocols.wireprotocol;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.reflect.TypeToken;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import org.corfudb.common.compression.Codec;
+import org.corfudb.protocols.logprotocol.CheckpointEntry;
+
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Utility class used by LogData to help serialize its contents.
+ */
+public class LogDataSerializerUtils {
+    // Prevent class from being instantiated
+    private LogDataSerializerUtils() {}
+
+    @FunctionalInterface
+    interface PayloadConstructor<T> {
+        T construct(ByteBuf buf);
+    }
+
+    @FunctionalInterface
+    interface Serializer {
+        void serialize(ByteBuf buffer, Object payload);
+    }
+
+    static ConcurrentHashMap<Class<?>, PayloadConstructor<?>> constructorMap = new ConcurrentHashMap<>(
+            ImmutableMap.<Class<?>, PayloadConstructor<?>>builder()
+                    .put(Byte.class, ByteBuf::readByte)
+                    .put(Integer.class, ByteBuf::readInt)
```

Comment:
Shouldn't this be an immutable map that is created once? Why do we add more types at runtime?